cmd/finch-daemon/router_utils.go

```diff
@@ -90,12 +93,47 @@ func createContainerdClient(conf *config.Config) (*backend.ContainerdClientWrapp
 	return backend.NewContainerdClientWrapper(client), nil
 }
 
+// sanitizeRegoFile validates and prepares the Rego policy file for use.
+// It checks validates the file, acquires a file lock,
+// and sets rego file to be read-only.
+func sanitizeRegoFile(options *DaemonOptions, logger *flog.Logrus) (string, error) {
+	if options.regoFilePath != "" {
+		if !options.enableMiddleware {
+			return "", fmt.Errorf("rego file path was provided without the --enable-middleware flag, please provide the --enable-middleware flag") // todo, can we default to setting this flag ourselves is this better UX?
+		}
+
+		if err := checkRegoFileValidity(options.regoFilePath); err != nil {
+			return "", err
+		}
+
+		if !options.skipRegoPermCheck {
+			fileInfo, err := os.Stat(options.regoFilePath)
+			if err != nil {
+				return "", fmt.Errorf("error checking rego file permissions: %v", err)
+			}
+
+			if fileInfo.Mode().Perm() > 0600 {
```

Comment:
This is not effective. `0477` < `0600` but should not be used. The check should be `fileInfo.Mode().Perm() & 0177 != 0`